src/MCPClient/lib/clientScripts/emailFailReport.py

```diff
@@ -20,166 +20,205 @@
-                databaseInterface.printErrors = True
-            except:
-                databaseInterface.printErrors = True
-                duration = 0
-                newRow.append(0)
-
-        rows2.append(newRow)
-
-    htmlcode2 = HTML.table(rows2, header_row=["Type", "Status", "Started"]) # TODO: Re-add duration
-    t2 = etree.fromstring(htmlcode2, parser).find("body/table")
-    i = 0  
-    for tr in t2.findall("tr"):
-        #header row
+    try:
+        server = smtplib.SMTP(server)
+        server.sendmail(efrom, to, msg.as_string())
+    except (smtplib.SMTPException):
+        logger.exception('Report email was not delivered')
+    finally:
+        server.quit()
+
+
+def get_file_stats(cursor, unit_type, unit_uuid):
+    sql = """
+        SELECT
+            COUNT(fileUUID)                        AS 'Number of files',
+            sum(fileSize)                          AS 'Total file size',
+            sum(fileSize)/count(fileUUID)/1000     AS 'Average file size KB',
+            sum(fileSize)/count(fileUUID)/1000000  AS 'Average file size MB'
+        FROM Files
+    """
+    if unit_type == 'sip':
+        sql += ' WHERE sipUUID = %s;'
+    elif unit_type == 'transfer':
+        sql += ' WHERE transferUUID = %s;'
+    cursor.execute(sql, [unit_uuid])
+    return cursor.fetchone()
+
+
+def get_processing_time(cursor, unit_type, unit_uuid):
+    sql = """
+        SELECT SEC_TO_TIME(SUM(TIMESTAMPDIFF(SECOND, Tasks.startTime, Tasks.endTime)))
+        FROM Tasks
+        LEFT JOIN Jobs ON Tasks.jobUUID = Jobs.jobUUID
+        WHERE Jobs.SIPUUID = %s
+    """
+    if unit_type == 'sip':
+        sql += ' AND Jobs.unitType IN ("unitSIP", "unitDIP");'
+    elif unit_type == 'transfer':
+        sql += ' AND Jobs.unitType IN ("unitTransfer");'
+    cursor.execute(sql, [unit_uuid])
+    return cursor.fetchone()
+
+
+def get_unit_statistical_data_html(unit_type, unit_uuid):
+    fields = [
+        'Unit type',
+        'Total time processing',
+        'Total file size',
+        'Number of files',
+        'Average file size KB',
+        'Average file size MB',
+    ]
+    values = (unit_type,)
+    unit_type = unit_type.lower()
+    if unit_type not in ('sip', 'transfer'):
+        raise ValueError('Unexpected value in unit_type: {}'.format(unit_type))
+    with connection.cursor() as cursor:
+        values += get_processing_time(cursor, unit_type, unit_uuid)
+        values += get_file_stats(cursor, unit_type, unit_uuid)
+    return HTML.table([values], header_row=fields)
+
+
+def get_unit_job_log_html(sip_uuid):
+    """
+    Generate a string containing a HTML table with the jobs of a given unit
+    identifier.
+    """
+    parser = etree.HTMLParser(remove_blank_text=True)
+    jobs = Job.objects.filter(sipuuid=sip_uuid, subjobof='').exclude(jobtype='Email fail report').order_by('-createdtime', '-createdtimedec')
+    rows = [(item.jobtype, item.currentstep, item.createdtime) for item in jobs]
```

Comment:
Tip: These two lines can be `rows = Job.objects.filter(sipuuid=sip_uuid, subjobof='').exclude(jobtype='Email fail report').order_by('-createdtime', '-createdtimedec').values_list('jobtype', 'currentstep', 'createdtime')`